### ISSUE TITLE
numpy arrays as inputs

### ISSUE THREAD

**Issue by fingoldo** [scikit-learn-contrib/categorical-encoding#131]
Does your package support this? I'd like to include ca in my regular sklearn pipelines, all of them are built using straight numpy arrays as inputs. However your documentation does not claim supporting it, neither any examples I was able to found. What am I missing? )

**Comment by wdm0006** [comment]
Numpy arrays are supported, but if you pass numpy arrays you have to specify columns with integer indices not with column names.

**Comment by janmotl** [comment]
Try to update to version 1.3.0, which was published yesterday. Something like this:
```
 pip3 install --upgrade category_encoders
```
works for me.

Weight of Evidence (WOE) is available only from 1.3.0. 

The mentioned line in TargetEncoder is not present in 1.3.0 anymore. Some basic tests of numpy inputs are in [test_encoders.py](https://github.com/scikit-learn-contrib/categorical-encoding/blob/master/category_encoders/tests/test_encoders.py). And following, just written test:
```python
    def test_np_col(self):
        for encoder_name in encoders.__all__:
            with self.subTest(encoder_name=encoder_name):
                # Encode a numpy array
                enc = getattr(encoders, encoder_name)(cols=[2,3,4,5])
                enc.fit(np_X, np_y)
                tu.verify_numeric(enc.transform(np_X_t))
```
passes for all encoders.

If the encoders still crash after the update, let us know.

**Comment by fingoldo** [comment]
Thank you so much! Indeed it works now ) Turned out I had v. 1.2.8, plus it was installed with some issues.  In 1.3.0 I can confirm integer indices and numpy arrays are working )
However, can you please add to documentation a mentioning that, in case of using Pandas dataframe as input, it can't have a string index? I've had hard times figuring out why both TargetEncoder and LeaveOneOurEncoders were returning nans... You can easily reproduce this in v 1.3.0 by using this code:

```python
import pandas as pd
from category_encoders import *
from sklearn.datasets import load_boston

# prepare some data
bunch = load_boston()
y = bunch.target
X = pd.DataFrame(bunch.data, columns=bunch.feature_names)

X.index=X.index.values.astype(str)

enc = TargetEncoder(cols=['CHAS', 'RAD']).fit(X, y)
numeric_dataset = enc.transform(X)
```

**Comment by janmotl** [comment]
Fixed with #150 (and previous changes) for all encoders
